experts-client/lib/experts-client.js

```diff
@@ -155,93 +237,107 @@ catch (err) {
+  async splay(cli) {
+
+    function str_or_file(opt,param,required) {
+      if (opt[param]) {
+        return opt[param];
+      } else if (opt[param+'@']) {
+        opt[param]=fs.readFileSync(opt[param+'@'],'utf8');
+        return opt[param];
+      } else if (required) {
+        console.error('missing required option: '+param+'(@)');
+        process.exit(1);
+      } else {
+        return null;
+      }
+    }
+
+//    console.log(cli);
+    const bind = str_or_file(cli,'bind',true);
+    const construct = str_or_file(cli,'construct',true);
+    const frame = str_or_file(cli,'frame',false)
+    if (cli.frame) {
+      cli.frame=JSON.parse(cli.frame);
+    }
+
+    let q;
+    let sources=null;
+    if (cli.quadstore) {
+      const db = await localDB.create({level:'ClassicLevel',path:cli.quadstore});
+      //  cli.source=[db];
+      q = new Engine(db.store);
+      sources=null;
     } else {
-      return null;
+      q = new QueryEngine();
+      sources=cli.source;
     }
-  }
-  
-  console.log(cli);
-  const bind = str_or_file(cli,'bind',true);
-  const construct = str_or_file(cli,'construct',true);
-  const frame = str_or_file(cli,'frame',false)
-  if (cli.frame) {
-    cli.frame=JSON.parse(cli.frame);
-  }
-  
-  let q;
-  let sources=null;
-  if (cli.quadstore) {
-    const db = await localDB.create({level:'ClassicLevel',path:cli.quadstore});
-    //  cli.source=[db];
-    q = new Engine(db.store);
-    sources=null;
-  } else {
-    q = new QueryEngine();
-    sources=cli.source;
-  }
-  
-  const factory=new DataFactory();
-  
-  const bindingStream=await q.queryBindings(cli.bind,{sources: cli.source})
-  bindingStream.on('data', construct_one )
-  .on('error', (error) => {
-    console.error(error);
-  })
-  .on('end', () => {
-    console.log('bindings done');
-  });
-
-async function construct_one(bindings) {
-  await bindingStream.off('data', construct_one);
-  let fn=1; // write to stdout by default
-  if ( bindings.get('filename') && bindings.get('filename').value) {
-    fn=bindings.get('filename').value
-  }
-  let graph = null;
-  if (bindings.get('graph')) {
-    graph=factory.namedNode(bindings.get('graph').value);
-  }
 
-  // convert construct to jsonld quads
-  const quadStream = await q.queryQuads(cli.construct,{initialBindings:bindings, sources: cli.source});
-  const quads = await quadStream.toArray();
-  if (graph) {
-    quads.forEach((quad) => {
-      quad.graph=graph;
-    });
-  }
-  console.log(`writing ${fn} with ${quads.length} quads`);
-  let doc=await jsonld.fromRDF(quads)
+    const factory=new DataFactory();
+
+    const bindingStream=await q.queryBindings(cli.bind,{sources: cli.source})
+    bindingStream.on('data', construct_one )
+      .on('error', (error) => {
+        console.error(error);
+      })
+      .on('end', () => {
+//        console.log('bindings done');
+      });
+
+    let binding_count=0;
+
+    async function construct_one(bindings) {
+      // binding_count++;
```

Comment:
This is definately the part that needs most work.  I removed this for my current example,  because the files to splay are small, but something like  this is needed for the IAM pull eg.